src/Aspire.Hosting/DistributedApplicationExecutionContext.cs

```diff
@@ -14,6 +14,17 @@ public class DistributedApplicationExecutionContext(DistributedApplicationOperat
     /// </summary>
     public DistributedApplicationOperation Operation { get; } = operation;
 
+    private IServiceProvider? _serviceProvider;
+
+    /// <summary>
+    /// The <see cref="IServiceProvider"/> for the AppHost.
+    /// </summary>
+    public IServiceProvider ServiceProvider
+    {
+        get =>_serviceProvider ?? throw new InvalidOperationException("The ServiceProvider has not been set.");
+        internal set => _serviceProvider = value;
```

Comment:
This probably won't be able to stay internal because it makes it impossible to unit test easily.

Reply:
It can't be internal, then you can't write tests for it.